pinot-core/src/main/java/org/apache/pinot/core/indexsegment/IndexSegment.java

```diff
@@ -57,6 +58,13 @@
    */
   Set<String> getPhysicalColumnNames();
 
+  /**
+   * Returns all columns for the "select *" query
+   *
+   * @return Set of column names
+   */
+  Set<String> getColumnNamesForSelectStar();
```

Comment:
We have two APIs, one named getColumnNames() and the other named getPhysicalColumnNames().  select * should decide which one to use? Do we need a third one? Why can't getPhysicalColumnNames API return the right set of columns from the consuming segment?

Reply:
For the consuming segment, `select *` should return physical columns + newly added columns. Since the newly added columns are provided by the virtual column provider, I feel it a bit weird to include them in getPhysicalColumnNames method. 

Basically the columns are
1. physical column
2. virtual column
 2.1 built-in virtual columns (docId, segmentName, hostName)
 2.2 newly added columns in the consuming segment

`getColumnNames` should return 1+2.1+2.2
`getPhysicalColumnNames` should return 1
`getColumnNamesForSelectStar` should return 1+2.2

Reply:
Why are we treating newly added columns as virtual?